packetbeat/protos/http/http.go

```diff
@@ -406,18 +406,33 @@ func (http *httpPlugin) handleHTTP(
 			debugf("Received response with tuple: %s", m.tcpTuple)
 		}
 		conn.responses.append(m)
-		http.correlate(conn)
+		http.correlate(conn, false)
 	}
 }
 
-func (http *httpPlugin) correlate(conn *httpConnectionData) {
-	// drop responses with missing requests
-	if conn.requests.empty() {
+func (http *httpPlugin) correlate(conn *httpConnectionData, flush bool) {
```

Comment:
Will the correlate always report all active messages? Depending on where packetbeat is running (passive sniffing), a request might be received after the corresponding resource. This make correlation quite tricky in general.

Is there a chance of flushRequests (at the end of the function) is prematurely flushing events? In which situation will the `flush` flag be set? Always on shutdown?

How about extracting the local functions into methods on the httpPlugin. The correlate can re-use them and we can remove the `flush` flag from correlate. Whenever we would set the flush flag, we would call `flushAll`, which calls `flushResponses` and `flushRequests`.